rwengine/src/ai/CharacterController.cpp

```diff
@@ -10,504 +10,450 @@
+  if (vehicle->vehicle->type == VehicleData::BOAT) {
+    character->enterVehicle(vehicle, seat);
+    return true;
+  }
+
+  if (seat == ANY_SEAT) {
+    // Determine which seat to take.
+    float nearest = std::numeric_limits<float>::max();
+    for (unsigned int s = 1; s < vehicle->info->seats.size(); ++s) {
+      auto entry = vehicle->getSeatEntryPositionWorld(s);
+      float dist = glm::distance(entry, character->getPosition());
+      if (dist < nearest) {
+        seat = s;
+        nearest = dist;
+      }
+    }
+  }
+
+  auto entryDoor = vehicle->getSeatEntryDoor(seat);
+  auto entryPos = vehicle->getSeatEntryPositionWorld(seat);
+  auto entryPosLocal = vehicle->getSeatEntryPosition(seat);
+
+  auto anm_open = character->animations.car_open_lhs;
+  auto anm_enter = character->animations.car_getin_lhs;
+  auto anm_pullout = character->animations.car_pullout_lhs;
+
+  if (entryPosLocal.x > 0.f) {
+    anm_open = character->animations.car_open_rhs;
+    anm_enter = character->animations.car_getin_rhs;
+    anm_pullout = character->animations.car_pullout_rhs;
+  }
+
+  // If there's someone in this seat already, we may have to ask them to leave.
+  auto currentOccupant = static_cast<CharacterObject *>(vehicle->getOccupant(seat));
+
+  bool tryToEnter = false;
+
+  if (entering) {
+    if (character->animator->getAnimation(AnimIndexAction) == anm_open) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        tryToEnter = true;
+      } else if (entryDoor && character->animator->getAnimationTime(AnimIndexAction) >= 0.5f) {
+        vehicle->setPartTarget(entryDoor, true, entryDoor->openAngle);
+      } else {
+        // character->setPosition(vehicle->getSeatEntryPosition(seat));
+        character->rotation = vehicle->getRotation();
+      }
+    } else if (character->animator->getAnimation(AnimIndexAction) == anm_pullout) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        tryToEnter = true;
+      }
+    } else if (character->animator->getAnimation(AnimIndexAction) == anm_enter) {
+      if (character->animator->isCompleted(AnimIndexAction)) {
+        // VehicleGetIn is over, finish activity
+        return true;
+      }
+    }
+  } else {
+    glm::vec3 targetDirection = entryPos - character->getPosition();
+    targetDirection.z = 0.f;
+
+    float targetDistance = glm::length(targetDirection);
+
+    if (targetDistance <= 0.4f) {
+      entering = true;
+      // Warp character to vehicle orientation
+      character->controller->setMoveDirection({0.f, 0.f, 0.f});
+      character->controller->setRunning(false);
+      character->setHeading(glm::degrees(glm::roll(vehicle->getRotation())));
+
+      // Determine if the door open animation should be skipped.
+      if (entryDoor == nullptr || (entryDoor->constraint != nullptr &&
+                                   glm::abs(entryDoor->constraint->getHingeAngle()) >= 0.6f)) {
```

Comment:
Using clang-format means you never think again about manually fixing
formatting issues - and you never fix source code manually.

The source code format is described by the clang-format rules and
formatting is part of a save action. You need to find a configuration for
clang-format containing your "personal" style.

After using auto formatting (with clang-format for c++ and eclipse/java) in
professional environments I never discussed with my team over source code
formatting again.


2016-08-19 12:59 GMT+02:00 Jannik Vogel <notifications@github.com>:

> In rwengine/src/ai/CharacterController.cpp
> <https://github.com/rwengine/openrw/pull/211#discussion_r75463639>:
>
> > +  } else {
> > +    glm::vec3 targetDirection = entryPos - character->getPosition();
> > +    targetDirection.z = 0.f;
> > +
> > +    float targetDistance = glm::length(targetDirection);
> > +
> > +    if (targetDistance <= 0.4f) {
> > +      entering = true;
> > +      // Warp character to vehicle orientation
> > +      character->controller->setMoveDirection({0.f, 0.f, 0.f});
> > +      character->controller->setRunning(false);
> > +      character->setHeading(glm::degrees(glm::roll(vehicle->getRotation())));
> > +
> > +      // Determine if the door open animation should be skipped.
> > +      if (entryDoor == nullptr || (entryDoor->constraint != nullptr &&
> > +                                   glm::abs(entryDoor->constraint->getHingeAngle()) >= 0.6f)) {
>
> This should have broken at the ||. However, we can fix these issues
> manually.
> Ideally we'd store more complicated logic in a boolean which also
> describes what's going on: bool isDoorOpened = ...
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rwengine/openrw/pull/211/files/23cd4799c29af8b04614bc56ab42dcadbc1b06bc#r75463639>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAO_BgcsE0cQqAhW7DBxzj5Ns6ZYMCpSks5qhYyagaJpZM4Jn97h>
> .
>